src/Symfony/Component/Serializer/Mapping/Factory/AbstractMetadataFactory.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Mapping\Factory;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+/**
+ * Abstract metadata factory.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+abstract class AbstractMetadataFactory implements ClassMetadataFactoryInterface
```

Comment:
this class does not seem useful in a class hierarchy. it looks more like a helper that should be implemented as a trait.